### ISSUE TITLE
Protobuf serialization fails for watching wallets

### ISSUE THREAD

**Issue by schildbach** [bitcoinj/bitcoinj#1209]
Consider this test case, to be pasted into WalletProtobufSerializerTest:

```
    @Test
    public void testRoundTripWatchingWallet() throws Exception {
        final String xpub = "tpubD9LrDvFDrB6wYNhbR2XcRRaT4yCa37TjBR3YthBQvrtEwEq6CKeEXUs3TppQd38rfxmxD1qLkC99iP3vKcKwLESSSYdFAftbrpuhSnsw6XM";
        final long creationTimeSeconds = 1457019819;
        Wallet wallet = Wallet.fromWatchingKey(PARAMS, DeterministicKey.deserializeB58(null, xpub, PARAMS),
                creationTimeSeconds);
        Wallet wallet2 = roundTrip(wallet);
        assertEquals(xpub, wallet.getWatchingKey().serializePubB58(PARAMS));
        assertEquals(xpub, wallet2.getWatchingKey().serializePubB58(PARAMS)); // fails!
        assertEquals(creationTimeSeconds, wallet.getWatchingKey().getCreationTimeSeconds()); // fails!
        assertEquals(creationTimeSeconds, wallet2.getWatchingKey().getCreationTimeSeconds()); // fails!
        assertEquals(creationTimeSeconds, wallet.getEarliestKeyCreationTime()); // fails!
        assertEquals(creationTimeSeconds, wallet2.getEarliestKeyCreationTime()); // fails!
    }
```

As you can see, bitcoinj fails to persist both the birth date and the xpub.

Pinging @devrandom and @oscarguindzberg as they were the last to touch the DKC serialization code.

**Comment by schildbach** [comment]
I think I fixed the issue in #1210. The root issue is a lot of the API puts the creation time into fields that can't get persisted to protobuf. See the fix for details, review is much appreciated.

**Comment by schildbach** [comment]
Fixed.
